bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandleAdv.java

```diff
@@ -241,7 +229,15 @@ public void safeRun() {
     @Override
     public CompletableFuture<Long> write(long entryId, ByteBuf data) {
         SyncAddCallback callback = new SyncAddCallback();
-        asyncAddEntry(entryId, data, callback, null);
+        PendingAddOp op = new PendingAddOp(this, callback, null);
```

Comment:
hmm... you just moved the logic back to write function... I still don't understand why do you need to duplicate the logic in `asyncAddEntry`. the new `write(long entryId, ByteBuf data)` should be just a shortcut of the legacy `asyncAddEntry` method, no?